src/prefect/tasks.py

```diff
@@ -756,7 +805,13 @@ def task(
     cache_key_fn: Callable[["TaskRunContext", Dict[str, Any]], Optional[str]] = None,
     cache_expiration: datetime.timedelta = None,
     retries: int = 0,
-    retry_delay_seconds: Union[float, int] = 0,
+    retry_delay_seconds: Union[
+        float,
+        int,
+        List[float],
+        Callable[["TaskRunContext", Dict[str, Any]], List[float]],
```

Comment:
Perhaps we should move this full `Union[...]` annotation into a variable e.g. `RetryDelayType = Union[...]` to avoid repeating this big union.

```suggestion
        Callable[[int], List[float]],
```